### PR TITLE
CODETOOLS-7903317: Fix code to compile extra property definitions

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review a fairly simple update to eliminate false positives in comments when looking for the name of the class in a Java source file.

The fix addresses a performance issue when running tests, because the "extra property definition" files were compiled every time jtreg was run, instead of as needed, which is the intent.

The fix is to ignore comments in the source code. The code to do this is copied from equivalent functionality in the langtools `ToolBox` class, which has been in use for many years.

The existing tests for "extra property definition" files have been extended with a test case to ensure that the false positives no longer occur and that the files are not recompiled unnecessarily.

Minor style issues in the main source file have also been fixed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [CODETOOLS-7903317](https://bugs.openjdk.org/browse/CODETOOLS-7903317): Fix code to compile extra property definitions

### Reviewers
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Daniel Jeliński](https://openjdk.org/census#djelinski) (@djelinski - no project role) ⚠️ Review applies to [e06107c0](https://git.openjdk.org/jtreg/pull/122/files/e06107c0d01d29e825c009976312eb8fb1a3e0d9)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jtreg pull/122/head:pull/122` \
`$ git checkout pull/122`

Update a local copy of the PR: \
`$ git checkout pull/122` \
`$ git pull https://git.openjdk.org/jtreg pull/122/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 122`

View PR using the GUI difftool: \
`$ git pr show -t 122`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jtreg/pull/122.diff">https://git.openjdk.org/jtreg/pull/122.diff</a>

</details>

### CODE REVIEW

File: test/extra-props/comments/TEST.ROOT [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2016, Oracle and/or its affiliates. All rights reserved.
```
Comment (irisclark) [inline test/extra-props/comments/TEST.ROOT:2]: Copyright year?

File: test/extra-props/comments/Test1.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2016, Oracle and/or its affiliates. All rights reserved.
```
Comment (irisclark) [inline test/extra-props/comments/Test1.java:2]: Copyright year?

File: test/extra-props/comments/Test1.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+/*
+ * @test
+ * @requires extra
+ */
+public class Test1 {
+    public static void main(String... args) {
+        // this test should be executed
+    }
+}
+
```
Comment (irisclark) [inline test/extra-props/comments/Test1.java:33]: extra trailing space

**Review by irisclark — APPROVED** [review]

File: src/share/classes/com/sun/javatest/regtest/config/ExtraPropDefns.java [hunk 4]
```diff
@@ -221,6 +223,9 @@ private List<String> compile(JDK jdk, Path classDir, SearchPath classpath,
         }
 
         if (needCompilation) {
+            if (trace) {
+                System.err.println("Compiling extra property definition files: " + javacArgs);
```
Comment (djelinski) [inline src/share/classes/com/sun/javatest/regtest/config/ExtraPropDefns.java:227]: Could you also log why the compilation was needed? i.e. when setting needCompilation, log sf and cf.

**Review by djelinski — APPROVED** [review]

File: src/share/classes/com/sun/javatest/regtest/config/ExtraPropDefns.java (hunk 4 above)
Comment (jonathan-gibbons) [reply src/share/classes/com/sun/javatest/regtest/config/ExtraPropDefns.java:227]: Will investigate, as a separate RFE

File: test/extra-props/comments/TEST.ROOT (hunk 1 above)
Comment (jonathan-gibbons) [reply test/extra-props/comments/TEST.ROOT:2]: Borderline, since it's a copy ...

**Review by irisclark — APPROVED** [review]

File: src/share/classes/com/sun/javatest/regtest/config/ExtraPropDefns.java (hunk 4 above)
Comment (jonathan-gibbons) [reply src/share/classes/com/sun/javatest/regtest/config/ExtraPropDefns.java:227]: https://bugs.openjdk.org/browse/CODETOOLS-7903329

**Pull request closed without merging**